chroma_core/lib/job.py

```diff
@@ -200,7 +200,21 @@ def invoke_rust_agent(self, host, command, args={}):
     def invoke_rust_agent_expect_result(self, host, command, args={}):
         from chroma_core.services.job_scheduler.agent_rpc import AgentException
 
-        result = json.loads(self.invoke_rust_agent(host, command, args))
+        try:
+            result = self.invoke_rust_agent(host, command, args)
+        except RustAgentCancellation as e:
+            raise AgentException(host, command, args, "Cancelled: {}".format(e))
+        except ValueError as e:
+            raise AgentException(host, command, args, result)
+        except Exception as e:
+            raise AgentException(host, command, args, "Unexpected error: {}".format(e))
+
+        try:
+            result = json.loads(result)
+        except ValueError as e:
+            raise AgentException(host, command, args, "Error parsing json: {}".format(e))
+        except Exception as e:
+            raise AgentException(host, command, args, "Unexpected error while parsing json: {}".format(e))
```

Comment:
Is there any other possible error besides a ValueError?

Reply:
I think we just nee ValueError. https://github.com/simplejson/simplejson/blob/v3.0.5/simplejson/decoder.py#L33 shows the the encoder exception inherits from ValueError. Should I remove the general exception underneath?

Reply:
Yes... unless there is a possibility Exception can be raised in addition to `ValueError`

Reply:
I did a quick search and din't see anything.